onnxruntime/core/framework/session_state.cc

```diff
@@ -258,45 +259,157 @@ void SessionState::CleanInitializedTensorsFromGraph() {
+
+                    const auto& op_type = node.OpType();
+
+                    // Sanity check
+                    // TODO: Check if some version of the ONNX IR allows op_type to be empty
+                    ORT_ENFORCE(!op_type.empty(), "The op type of a node cannot be empty");
+
+                    std::ostringstream ss_1;
+                    ss_1 << op_type;
+                    ss_1 << "+";
+                    ss_1 << std::to_string(prepacked_weight_hash);
+
+                    // The key for the pre-packed weights container lookup is the op_type + hash of the prepacked-weight
+                    // that we just got by invoking PrePack() on this kernel.
+
+                    const std::string& prepacked_weights_container_key = ss_1.str();
+
+                    bool container_contains_packed_weight = prepacked_weights_container_->HasCachedWeight(prepacked_weights_container_key);
+
+                    if (container_contains_packed_weight) {
+                      LOGS(logger_, INFO) << "Using cached version of pre-packed weight for constant initializer: " << input_name
+                                          << " used in the node: " << node.Name() << " which is of op type: " << node.OpType();
+
+                      ORT_RETURN_IF_ERROR(KernelStorePrePackedWeight(kernel, input_idx,
+                                                                     prepacked_weights_container_->GetCachedWeight(prepacked_weights_container_key),
+                                                                     node.Name()));
+
+                      ++used_cached_pre_packed_weights_counter_;
+                    } else {  // container doesn't contain the pre-packed weight - so write into it
+
+                      // The cache has (atleast one) pre-packed weight corresponding to this op_type + shared initializer combo,
+                      // but the pre-packed weight computed using this instance of the op_type and the same shared initializer has different contents
+                      // than what exists in the cache. This is possible when the contents of the pre-packed buffer is a function of other
+                      // static attributes of the node - for example, the node's attributes, the model's opset version etc.
+                      // We log this occurrence for the user.
+                      if (prepacked_weights_container_->HasPrepackedWeightForOpTypeAndConstantInitializer(op_type, const_initialized_tensor.DataRaw())) {
+                        LOGS(logger_, WARNING) << "The pre-packed weights container has (atleast one) pre-packed weight corresponding to this op_type: " << node.Name()
+                                               << " and this shared initializer but the pre-packed weight computed using this instance of the op_type and the"
+                                                  " same shared initializer has different contents than what exists in the cache";
+                      }
+
```

Comment:
Why do we log this warning? We store the correct values don't we, so why would the user care and what can they do about it even if they did?

Reply:
We log this warning as a debug message - basically we would like to know if the same shared initializer somehow resulted in different PrePackedWeights - this is possible if a kernel's PrePack() implementation took a dependency on static node properties like node attributes along with the shared initializer's contents to define the contents of the pre-packed buffer.

Maybe we could adjust logging level/make it a debug log ?

Reply:
Yeah, the user doesn't care about this. Can be logged at debug level.

Reply:
This log goes away with the recommendation that we don't need to track this at all.